kasa/smartstrip.py

```diff
@@ -227,9 +233,29 @@ async def _query_helper(
         self, target: str, cmd: str, arg: Optional[Dict] = None, child_ids=None
     ) -> Any:
         """Override query helper to include the child_ids."""
-        return await self.parent._query_helper(
-            target, cmd, arg, child_ids=[self.child_id]
-        )
+        # the discovery protocol only returns the index for the child ID
+        #  but the 2021 KP400 requires the full ID including the parent ID
+        #  so we try both
+        child_id = self.child_id
+        child_ids = [child_id]
+        if len(child_id) < 3:
+            child_ids.append(self.parent.sys_info["deviceId"] + child_id)
+        retval = None
+        e: Exception = None  # type: ignore
+        for id in child_ids:
+            try:
+                retval = await self.parent._query_helper(
+                    target, cmd, arg, child_ids=[id]
+                )
+                return retval
+            except Exception as ex:
+                # swallow this exception and try the other way
+                e = ex
+                next
+
+        # the only way to get here is if both attempts to communicate with the child failed
+        # so we pass along the latest exception
+        raise e
```

Comment:
This looks quite convoluted. Would it be possible to simplify this and do the detection only once, either inside `create_children` (and pass the correct ID to `SmartStripPlug`?), or if an error happens during the first invokation?

Reply:
agree. I simplified it to just catch the exception to update the child_id and try  again.